sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/AggregateOptimizeSuite.scala

```diff
@@ -71,4 +74,33 @@ class AggregateOptimizeSuite extends AnalysisTest {
 
     comparePlans(optimized, correctAnswer)
   }
+
+  test("Remove left join if it only has distinct on left side") {
```

Comment:
Although it's not required for new features, could you add a test prefix, `SPARK-34808:`, for reviewers, please?